lib/chai/interface/assert.js

```diff
@@ -2767,8 +2766,8 @@ module.exports = function (chai, util) {
   assert.decreasesBy = function (fn, obj, prop, delta, msg) {
     if (arguments.length === 4 && typeof obj === 'function') {
       var tmpMsg = delta;
-      delta = prop;
       msg = tmpMsg;
+      prop = null;
```

Comment:
I think this is not correct.

When we have four arguments and the type of `obj` is a function, we cannot set `prop` to null because the user might desire to check for the value of a `prop` in a `function`, since a `function` in JavaScript can have properties.

Please correct me if I'm wrong.

Reply:
I will try to exclude this commit, focusing on testing only. Either we both wrong or none of us is wrong.